simplified-app-shared/src/main/java/org/nypl/simplified/app/reader/ReaderTOCBookmarksFragment.kt

```diff
@@ -1,18 +1,132 @@
 package org.nypl.simplified.app.reader
 
-import android.graphics.Color
+import android.content.Context
+import android.database.DataSetObserver
 import android.os.Bundle
 import android.support.v4.app.Fragment
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import android.widget.*
+
 import org.nypl.simplified.app.R
+import org.nypl.simplified.app.Simplified
+
+/**
+ * A reusable fragment for a ListView of bookmarks
+ */
+
+class ReaderTOCBookmarksFragment : Fragment(), ListAdapter {
+
+  private var inflater: LayoutInflater? = null
+  private var adapter: ArrayAdapter<ReaderTOC.TOCElement>? = null
+  private var listener: ReaderTOCBookmarksFragmentSelectionListenerType? = null
+
+  private var bookmarksTOCLayout: View? = null
+  private var bookmarksTOCListView: ListView? = null
 
-class ReaderTOCBookmarksFragment : Fragment() {
 
   override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                             savedInstanceState: Bundle?): View? {
 
-    return inflater.inflate(R.layout.reader_toc_bookmarks, null)
+    this.inflater = inflater
+    bookmarksTOCLayout = inflater.inflate(R.layout.reader_toc_bookmarks, null)
+    bookmarksTOCListView = bookmarksTOCLayout?.findViewById(R.id.bookmarks_list) as? ListView
+
+    val activity = activity as? ReaderTOCActivity
+
+    //TODO WIP: change to bookmarks
+    val listViewElements = activity?.in_toc?.elements as? List<ReaderTOC.TOCElement>
+    adapter = ArrayAdapter(context, 0, listViewElements)
+    bookmarksTOCListView?.adapter = this
+
+    //TODO ignoring color scheme listener stuff from contents fragment
+
+    return bookmarksTOCLayout
+  }
+
+  override fun onAttach(context: Context) {
+    super.onAttach(context)
+    if (context is ReaderTOCBookmarksFragmentSelectionListenerType) {
+      listener = context
+    } else {
+      throw RuntimeException(context.toString() +
+          " must implement ReaderTOCBookmarksFragmentSelectionListenerType")
+    }
+  }
+
+  /**
+   * List View Adapter
+   */
+
+  override fun areAllItemsEnabled(): Boolean {
+    return adapter!!.areAllItemsEnabled()
+  }
+
+  override fun getCount(): Int {
+    return adapter!!.count
+  }
+
+  override fun getItem(position: Int): Any {
+    return adapter!!.getItem(position)
+  }
+
+  override fun getItemId(position: Int): Long {
+    return adapter!!.getItemId(position)
+  }
+
+  override fun getItemViewType(position: Int): Int {
+    return adapter!!.getItemViewType(position)
```

Comment:
is there a way we can provide default values here instead of hitting null pointer exceptions? 

https://kotlinlang.org/docs/reference/null-safety.html#safe-calls

Reply:
I am referring to all uses of `adapter!!` in this class

Reply:
I'll make an issue for this so I don't forget.